src/linker/Linker.Steps/MethodBodyInstructionsProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker.Steps
+{
+	public class MethodBodyInstructionsProvider
+	{
+		readonly UnreachableBlocksOptimizer _unreachableBlocksOptimizer;
+
+		public readonly struct ProcessedMethodBody
```

Comment:
This is also something I'd like to get feedback on. Basically in the MarkStep/MethodBodyScanner there's a lot of code which accesses instructions but all of it happens in code paths which are guaranteed to have the method correctly processed already. I didn't want to add bunch of warning suppressions all of the codebase as it's messy and hard to prove that it's correct. Instead I introduced this wrapper struct which represented a method which is "Safe" to access instructions on. This means the struct is now passed around quite a bit.

So mainly looking for two things:
* Opinions about this solution + alternative ideas
* Naming suggestions/code factoring - even if we keep this, maybe we should rename/move it somehow

Reply:
`MethodIL` is not ideal - but it's probably better than "processed". While I agree "processed" is more accurate, the consumers of this "don't care".

Reply:
With the move to `LinkContext` I also moved this. There is now `MethodIL` type in the `Mono.Linker` namespace - and the rest is on `LinkContext`.